src/content/language/isolates.md

```diff
@@ -39,6 +39,12 @@ but here are some more situations where they can be useful:
 - Performing I/O, such as communicating with a database.
 - Handling a large volume of network requests.
 
+Also note a caveat due to a limit to the number of concurrently running isolates:
+
+ - The limit is not hardcoded to a particular number, it is calculated based on the Dart VM heap size available to the Dart application, can be considered to be between 8 and 32 depending on the platform.
+ - This limit doesn't affect asynchronous communction between isolates via messages - you can have hundreds of isolates running and making progress. The isolates are scheduled on the CPU in round-robin fashion and yield to each other often.
+ - Attempts to do *synchronous* communication between isolates over the limit though may result in a deadlock.
```

Comment:
... unless special care is taken (the C code that does synchronous communication would need to leave the current isolate before it blocks and re-enter it before returning to dart, see ...)

Reply:
Thanks for the comment, I added this. PTAL.